vm-virtio/src/transport/pci_device.rs

```diff
@@ -239,6 +239,9 @@ pub struct VirtioPciDevice {
 
     // Setting PCI BAR
     settings_bar: u8,
+
```

Comment:
Just some general questions about your commit message here. You said:
```
Currently the hypervisor sets up an identify mapping of 4GiB which the firmware extends to 64GiB
```
- Why is the firmware actually extending from 4G to 64G?
- Why is this extension needed?
- Why 64G and not 32G or 128G for instance?

Reply:
The hypervisor only creates an identity mapping for 1GiB (not 4GiB, sorry my mistake). You need the extension as you always needs a identity page table entries to access any memory. Previously with our 64GiB address space limit that would require extending to 64GiB to be able to access the PCI devices as they were put at the top of that address space.

Reply:
>Previously with our 64GiB address space limit that would require extending to 64GiB to be able to access the PCI devices as they were put at the top of that address space.

Yes so if we end up with an address space of 1TiB, we need the extension to 1TiB now, right? I mean your patch didn't change the behavior about where to place the devices (they're still at the end of the address space), except for the virtio-blk one.

Reply:
We don't need it because the block devices are in the 32-bit gap (this was the whole point of moving the block devices.)

Reply:
Ah okay now I understand what you meant. We were extending to 64GiB, but we would have needed to extend to even bigger when we started to look at supporting real physical address space, and instead we simplified the whole problem by moving block and not extending at all in the firmware.